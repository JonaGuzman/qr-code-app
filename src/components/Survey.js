import React from "react";
import QuestionAndAnswer from "./QuestionAndAnswer";

export default class Survey extends React.Component {
  constructor() {
    super();
    this.handleSubmit = this.handleSubmit.bind(this);
    this.state = {
      title: "Poster 1",
      id: 1,
      qr_id: "332434%fgd",
      user: "test@test.com",
      comment: "",
      question_id: 1,
      question: "Why am I here?",
      answer: ""
    };
  }

  handleSubmit = e => {
    e.preventDefault();
    const url = "https://jsonplaceholder.typicode.com/posts";
    fetch(url, {
      method: "POST",
      body: JSON.stringify(this.state),
      headers: { "Content-type": "application/json" }
    })
      .then(res => res.json())
      .catch(error => console.error("Error:", error))
      .then(response => console.log("Success:", response));
  };

  onTextChange = e => {
    const answer = e.target.value;
    this.setState(() => ( { answer } ));
  };

  onUpdateComment = e => {
    const comment = e.target.value;
    this.setState(() => ( { comment }));
  };
  
  render() {
    return (
      <div className="container-fluid">
        <h1 className="mx-auto">{this.state.title}</h1>
        <form 
          onSubmit={this.handleSubmit}
          style={{display: 'grid', gridGap: '1rem', justifyContent: 'center'}}>
          <div className="form-group">
          <label>User:</label>
            <input
              className="form-control"
              type="email"
              placeholder="Email"
              defaultValue={this.state.user}
            />
              <QuestionAndAnswer
                key={this.state.question_id}
                question={this.state.question}
                value={this.state.answer}
                handleAnswerEvent={this.onTextChange.bind(this)} 
              />
            <label>Comment:</label>
            <textarea className="form-control" rows="2" onChange={this.onUpdateComment.bind(this)}></textarea>
          </div>
          <button>Submit</button>
        </form>
      </div>
    );
  }
}
