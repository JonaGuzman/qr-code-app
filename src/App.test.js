import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Survey from './components/Survey';

test('Title Poster 1 Appears', () => {
  const { getByText } = render(<Survey />);
  const posterName = getByText(/Poster 1/i);
  expect(posterName).toBeInTheDocument();
});

test('Survey Snapshot Test', () => {
  const survey = renderer.create(<Survey />).toJSON();
  expect(survey).toMatchSnapshot();
});
