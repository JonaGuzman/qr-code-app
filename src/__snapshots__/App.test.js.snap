// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey Snapshot Test 1`] = `
<div
  className="container-fluid"
>
  <h1
    className="mx-auto"
  >
    Poster 1
  </h1>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "display": "grid",
        "gridGap": "1rem",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="form-group"
    >
      <div>
        <label>
          User:
        </label>
        <input
          className="form-control"
          defaultValue="test@test.com"
          name="Email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
      <div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={1}
            >
              Why?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={1}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={2}
            >
              Who?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={2}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={3}
            >
              How?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={3}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={4}
            >
              When?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={4}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={5}
            >
              What?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={5}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={6}
            >
              Where?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={6}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="formGroup"
          style={
            Object {
              "padding": "1rem 0",
            }
          }
        >
          <div>
            <label
              id={7}
            >
              Why is that?
            </label>
          </div>
          <div>
            <input
              className="form-control"
              id={7}
              onChange={[Function]}
              placeholder="Answer"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <label>
          Comment:
        </label>
        <textarea
          className="form-control"
          name="Comment"
          onChange={[Function]}
          rows="2"
        />
      </div>
    </div>
    <button
      className="btn btn-primary btn-lg"
    >
      Submit
    </button>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "paddingBottom": "1rem",
          "textAlign": "center",
        }
      }
    >
      <a
        href=""
        style={
          Object {
            "padding": "1rem",
          }
        }
      >
        <canvas
          height={128}
          id="332434%fgd"
          style={
            Object {
              "height": 128,
              "width": 128,
            }
          }
          width={128}
        />
      </a>
      <button
        className="btn btn-primary btn-sm"
        onClick={[Function]}
        type="button"
      >
         Download QR 
      </button>
    </div>
  </form>
</div>
`;
